src/sonic-bgpcfgd/bgpcfgd/managers_bgp.py

```diff
@@ -129,9 +129,6 @@ def __init__(self, common_objs, db_name, table_name, peer_type, check_neig_meta)
         if self.check_deployment_id:
             deps.append(("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, "localhost/deployment_id"))
 
-        if self.peer_type == 'internal':
-            deps.append(("CONFIG_DB", swsscommon.CFG_LOOPBACK_INTERFACE_TABLE_NAME, "Loopback4096"))
```

Comment:
why we remove this change ?

Reply:
This dep is used to decide whether to invoke set_handler for `managers_bgp` [managers.py](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-bgpcfgd/bgpcfgd/manager.py#L41), when subscribed table changed, if deps doesn't exist, it will not invoke set hanlder and put it into a queue, check again when subscribed change next time (for `managers_bgp`, this handler is to add bgp peer). We moved this hard dependency into add function [managers_bgp.py](https://github.com/sonic-net/sonic-buildimage/blob/ecdbf44fd1c9f91ab7b673122effe37a77035d42/src/sonic-bgpcfgd/bgpcfgd/managers_bgp.py#L174), because we can accept that Loopback4096 doesn't have ipv4 address if bgp_router_id configured. To be short, this dep is only a condition for whether to add bgp peer, we moved it into add_peer function. **It wouldn't affect table subscription, bgpcfgd will still monitor LOOPBACK_INTERFACE table and invoke corresponding handler [main.py](https://github.com/sonic-net/sonic-buildimage/blob/ecdbf44fd1c9f91ab7b673122effe37a77035d42/src/sonic-bgpcfgd/bgpcfgd/main.py#L50), we didn't change this part**